File: scripts/fix_accessibility.py
#!/usr/bin/env python3
"""
Script para corrigir TODOS os accessibility issues automaticamente.
Foca em adicionar contentDescription em Icons, Images e onClickLabel em Clickables.
"""

import os
import re
import sys

# Mapeamento de ícones para content descriptions
ICON_MAP = {
    'ArrowBack': 'cd_back',
    'ArrowForward': 'cd_forward',
    'Close': 'cd_close',
    'Menu': 'cd_menu',
    'MoreVert': 'cd_more_options',
    'MoreHoriz': 'cd_more_options',
    'Add': 'cd_add',
    'Delete': 'cd_delete',
    'Edit': 'cd_edit',
    'Save': 'cd_save',
    'Share': 'cd_share',
    'Search': 'cd_search',
    'Filter': 'cd_filter',
    'Check': 'cd_check',
    'CheckCircle': 'cd_check_circle',
    'Cancel': 'cd_cancel',
    'Clear': 'cd_clear',
    'Refresh': 'cd_refresh',
    'Send': 'cd_send',
    'Star': 'cd_star',
    'Favorite': 'cd_favorite',
    'FavoriteBorder': 'cd_favorite_border',
    'Person': 'cd_person',
    'Group': 'cd_group',
    'Groups': 'cd_groups',
    'LocationOn': 'cd_location',
    'Place': 'cd_place',
    'Event': 'cd_event',
    'CalendarToday': 'cd_calendar',
    'Schedule': 'cd_schedule',
    'SportsSoccer': 'cd_soccer',
    'EmojiEvents': 'cd_trophy',
    'Info': 'cd_info',
    'Help': 'cd_help',
    'Warning': 'cd_warning',
    'Error': 'cd_error',
    'Settings': 'cd_settings',
    'Notifications': 'cd_notifications',
    'Visibility': 'cd_show',
    'VisibilityOff': 'cd_hide',
    'KeyboardArrowDown': 'cd_expand',
    'KeyboardArrowUp': 'cd_collapse',
    'ExpandMore': 'cd_expand',
    'ExpandLess': 'cd_collapse',
    'ChevronRight': 'cd_chevron_right',
    'ChevronLeft': 'cd_chevron_left',
    'AccountCircle': 'cd_account',
    'ExitToApp': 'cd_logout',
    'AttachMoney': 'cd_money',
    'QrCode': 'cd_qr_code',
    'Camera': 'cd_camera',
    'Image': 'cd_image',
    'Phone': 'cd_phone',
    'Email': 'cd_email',
}


def get_icon_cd_key(icon_name):
    """Get contentDescription key for an icon."""
    return ICON_MAP.get(icon_name, 'cd_icon')


def fix_icon_calls(content):
    """Fix Icon() calls that are missing contentDescription."""
    modified = False

    # Pattern 1: Icon(Icons.xxx) - simple case
    def replace_simple_icon(match):
        nonlocal modified
        icon_full = match.group(1)
        icon_name = icon_full.split('.')[-1]
        cd_key = get_icon_cd_key(icon_name)
        modified = True
        return f'Icon({icon_full}, contentDescription = stringResource(R.string.{cd_key}))'

    content = re.sub(
        r'Icon\s*\(\s*(Icons\.[A-Za-z.]+)\s*\)',
        replace_simple_icon,
        content
    )

    # Pattern 2: Icon(imageVector = Icons.xxx, ...) without contentDescription
    lines = content.split('\n')
    new_lines = []
    i = 0
    while i < len(lines):
        line = lines[i]

        # Check if this line starts an Icon call
        if re.match(r'\s*Icon\s*\(', line):
            # Collect the full Icon call
            icon_call_lines = [line]
            paren_count = line.count('(') - line.count(')')
            j = i + 1

            while paren_count > 0 and j < len(lines):
                icon_call_lines.append(lines[j])
                paren_count += lines[j].count('(') - lines[j].count(')')
                j += 1

            # Check if contentDescription is already present
            icon_call_text = '\n'.join(icon_call_lines)

            if 'contentDescription' not in icon_call_text:
                # Extract icon name if present
                icon_match = re.search(r'Icons\.[A-Za-z.]+\.(\w+)', icon_call_text)
                if icon_match:
                    icon_name = icon_match.group(1)
                    cd_key = get_icon_cd_key(icon_name)

                    # Find where to insert contentDescription
                    # Insert after imageVector line if present
                    for k, call_line in enumerate(icon_call_lines):
                        if 'imageVector' in call_line and k + 1 < len(icon_call_lines):
                            indent = len(call_line) - len(call_line.lstrip())
                            cd_line = ' ' * indent + f'contentDescription = stringResource(R.string.{cd_key}),'
                            icon_call_lines.insert(k + 1, cd_line)
                            modified = True
                            break

                    new_lines.extend(icon_call_lines)
                    i = j
                    continue

            new_lines.extend(icon_call_lines)
            i = j
        else:
            new_lines.append(line)
            i += 1

    if modified:
        content = '\n'.join(new_lines)

    return content, modified


def fix_async_images(content):
    """Fix AsyncImage calls that are missing contentDescription."""
    modified = False

    def replace_async_image(match):
        nonlocal modified
        full_match = match.group(0)
        if 'contentDescription' not in full_match:
            modified = True
            # Insert contentDescription after model parameter
            return re.sub(
                r'(model\s*=\s*[^,]+,)',
                r'\1\n    contentDescription = stringResource(R.string.cd_profile_photo),',
                full_match
            )
        return full_match

    content = re.sub(
        r'AsyncImage\s*\([^)]+\)',
        replace_async_image,
        content,
        flags=re.DOTALL
    )

    return content, modified


def fix_clickables(content):
    """Fix .clickable calls that are missing onClickLabel."""
    modified = False

    def replace_clickable(match):
        nonlocal modified
        full_match = match.group(0)
        if 'onClickLabel' not in full_match:
            modified = True
            return '.clickable(\n        onClickLabel = stringResource(R.string.action_click)\n    ) {'
        return full_match

    content = re.sub(
        r'\.clickable\s*\{',
        replace_clickable,
        content
    )

    return content, modified


def add_import_if_needed(content, modified):
    """Add stringResource import if not present and modifications were made."""
    if modified and 'import androidx.compose.ui.res.stringResource' not in content:
        # Find the imports section and add
        import_match = re.search(r'(import androidx\.compose[^\n]*\n)', content)
        if import_match:
            content = content.replace(
                import_match.group(0),
                import_match.group(0) + 'import androidx.compose.ui.res.stringResource\n'
            )
        else:
            # Add after package declaration
            package_match = re.search(r'(package [^\n]+\n\n)', content)
            if package_match:
                content = content.replace(
                    package_match.group(0),
                    package_match.group(0) + 'import androidx.compose.ui.res.stringResource\n\n'
                )

    return content


def process_file(file_path):
    """Process a single Kotlin file."""
    # Skip non-UI files
    if any(x in file_path for x in ['/model/', '/data/', '/domain/', '/util/', 'FcmService']):
        return False

    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()

        original = content
        overall_modified = False

        # Fix Icons
        content, mod1 = fix_icon_calls(content)
        overall_modified = overall_modified or mod1

        # Fix AsyncImages
        content, mod2 = fix_async_images(content)
        overall_modified = overall_modified or mod2

        # Fix Clickables
        content, mod3 = fix_clickables(content)
        overall_modified = overall_modified or mod3

        # Add import if needed
        if overall_modified:
            content = add_import_if_needed(content, overall_modified)

        if overall_modified:
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(content)
            return True

        return False

    except Exception as e:
        print(f'Error processing {file_path}: {e}', file=sys.stderr)
        return False


def get_all_kotlin_files(directory):
    """Recursively get all Kotlin files in directory."""
    kotlin_files = []
    for root, dirs, files in os.walk(directory):
        for file in files:
            if file.endswith('.kt'):
                kotlin_files.append(os.path.join(root, file))
    return kotlin_files


def main():
    print('>>> Accessibility Fix - Python Edition\n')

    # Get project root
    script_dir = os.path.dirname(os.path.abspath(__file__))
    project_root = os.path.dirname(script_dir)
    app_dir = os.path.join(project_root, 'app', 'src', 'main', 'java')

    # Get all Kotlin files
    kotlin_files = get_all_kotlin_files(app_dir)
    print(f'Found {len(kotlin_files)} Kotlin files\n')

    # Process files
    modified_count = 0
    for i, file_path in enumerate(kotlin_files):
        if process_file(file_path):
            modified_count += 1
            rel_path = os.path.relpath(file_path, app_dir)
            print(f'[OK] {rel_path}')

        if (i + 1) % 100 == 0:
            print(f'Progress: {i + 1}/{len(kotlin_files)}...')

    print(f'\n>>> Modified {modified_count}/{len(kotlin_files)} files')
    print('\nRun: ./gradlew compileDebugKotlin to verify')


if __name__ == '__main__':
    main()

File: scripts/fix-accessibility.js
/**
 * Script para corrigir automaticamente todos os issues de acessibilidade
 * Adiciona contentDescription em Icons, Images e onClickLabel em Clickables
 */

const fs = require('fs');
const path = require('path');

// Mapeamento de ícones para contentDescription
const iconDescriptions = {
  // Navigation
  'ArrowBack': 'cd_back',
  'ArrowForward': 'cd_forward',
  'Close': 'cd_close',
  'Menu': 'cd_menu',
  'MoreVert': 'cd_more_options',
  'MoreHoriz': 'cd_more_options',

  // Actions
  'Add': 'cd_add',
  'Delete': 'cd_delete',
  'Edit': 'cd_edit',
  'Save': 'cd_save',
  'Share': 'cd_share',
  'Search': 'cd_search',
  'Filter': 'cd_filter',
  'Check': 'cd_check',
  'CheckCircle': 'cd_check_circle',
  'Cancel': 'cd_cancel',
  'Clear': 'cd_clear',
  'Refresh': 'cd_refresh',
  'Send': 'cd_send',

  // Content
  'Star': 'cd_star',
  'Favorite': 'cd_favorite',
  'FavoriteBorder': 'cd_favorite_border',
  'Person': 'cd_person',
  'Group': 'cd_group',
  'Groups': 'cd_groups',
  'LocationOn': 'cd_location',
  'Place': 'cd_place',
  'Event': 'cd_event',
  'CalendarToday': 'cd_calendar',
  'Schedule': 'cd_schedule',
  'Timer': 'cd_timer',
  'AccessTime': 'cd_time',
  'DateRange': 'cd_date_range',

  // Sports
  'SportsSoccer': 'cd_soccer',
  'EmojiEvents': 'cd_trophy',
  'Celebration': 'cd_celebration',
  'MilitaryTech': 'cd_badge',
  'Star': 'cd_star',
  'WorkspacePremium': 'cd_premium',

  // Info
  'Info': 'cd_info',
  'InfoOutlined': 'cd_info',
  'Help': 'cd_help',
  'Warning': 'cd_warning',
  'Error': 'cd_error',
  'ErrorOutline': 'cd_error',

  // Settings
  'Settings': 'cd_settings',
  'Tune': 'cd_tune',
  'Notifications': 'cd_notifications',
  'NotificationsNone': 'cd_notifications',

  // Visibility
  'Visibility': 'cd_show',
  'VisibilityOff': 'cd_hide',

  // Navigation arrows
  'KeyboardArrowDown': 'cd_expand',
  'KeyboardArrowUp': 'cd_collapse',
  'KeyboardArrowRight': 'cd_next',
  'KeyboardArrowLeft': 'cd_previous',
  'ExpandMore': 'cd_expand',
  'ExpandLess': 'cd_collapse',
  'ChevronRight': 'cd_chevron_right',
  'ChevronLeft': 'cd_chevron_left',

  // Social
  'Chat': 'cd_chat',
  'Comment': 'cd_comment',
  'ThumbUp': 'cd_like',
  'ThumbDown': 'cd_dislike',

  // Media
  'Image': 'cd_image',
  'Photo': 'cd_photo',
  'Camera': 'cd_camera',
  'CameraAlt': 'cd_camera',

  // Status
  'CheckCircle': 'cd_completed',
  'RadioButtonUnchecked': 'cd_unchecked',
  'Circle': 'cd_indicator',

  // Money
  'AttachMoney': 'cd_money',
  'Payment': 'cd_payment',
  'AccountBalance': 'cd_balance',

  // Misc
  'QrCode': 'cd_qr_code',
  'QrCodeScanner': 'cd_qr_scanner',
  'Download': 'cd_download',
  'Upload': 'cd_upload',
  'Link': 'cd_link',
  'ContentCopy': 'cd_copy',
  'ContentPaste': 'cd_paste',
  'Phone': 'cd_phone',
  'Email': 'cd_email',
  'Lock': 'cd_lock',
  'LockOpen': 'cd_unlock',
  'VpnKey': 'cd_key',
  'AccountCircle': 'cd_account',
  'ExitToApp': 'cd_logout',
  'Login': 'cd_login',
  'Logout': 'cd_logout',
};

// Strings para adicionar ao strings.xml
const contentDescriptionStrings = {
  // Navigation
  'cd_back': 'Back',
  'cd_forward': 'Forward',
  'cd_close': 'Close',
  'cd_menu': 'Menu',
  'cd_more_options': 'More options',

  // Actions
  'cd_add': 'Add',
  'cd_delete': 'Delete',
  'cd_edit': 'Edit',
  'cd_save': 'Save',
  'cd_share': 'Share',
  'cd_search': 'Search',
  'cd_filter': 'Filter',
  'cd_check': 'Check',
  'cd_check_circle': 'Checked',
  'cd_cancel': 'Cancel',
  'cd_clear': 'Clear',
  'cd_refresh': 'Refresh',
  'cd_send': 'Send',

  // Content
  'cd_star': 'Star',
  'cd_favorite': 'Favorite',
  'cd_favorite_border': 'Add to favorites',
  'cd_person': 'Person',
  'cd_group': 'Group',
  'cd_groups': 'Groups',
  'cd_location': 'Location',
  'cd_place': 'Place',
  'cd_event': 'Event',
  'cd_calendar': 'Calendar',
  'cd_schedule': 'Schedule',
  'cd_timer': 'Timer',
  'cd_time': 'Time',
  'cd_date_range': 'Date range',

  // Sports
  'cd_soccer': 'Soccer',
  'cd_trophy': 'Trophy',
  'cd_celebration': 'Celebration',
  'cd_badge': 'Badge',
  'cd_premium': 'Premium',

  // Info
  'cd_info': 'Information',
  'cd_help': 'Help',
  'cd_warning': 'Warning',
  'cd_error': 'Error',

  // Settings
  'cd_settings': 'Settings',
  'cd_tune': 'Tune',
  'cd_notifications': 'Notifications',

  // Visibility
  'cd_show': 'Show',
  'cd_hide': 'Hide',

  // Navigation arrows
  'cd_expand': 'Expand',
  'cd_collapse': 'Collapse',
  'cd_next': 'Next',
  'cd_previous': 'Previous',
  'cd_chevron_right': 'Navigate forward',
  'cd_chevron_left': 'Navigate back',

  // Social
  'cd_chat': 'Chat',
  'cd_comment': 'Comment',
  'cd_like': 'Like',
  'cd_dislike': 'Dislike',

  // Media
  'cd_image': 'Image',
  'cd_photo': 'Photo',
  'cd_camera': 'Camera',

  // Status
  'cd_completed': 'Completed',
  'cd_unchecked': 'Unchecked',
  'cd_indicator': 'Indicator',

  // Money
  'cd_money': 'Money',
  'cd_payment': 'Payment',
  'cd_balance': 'Balance',

  // Misc
  'cd_qr_code': 'QR Code',
  'cd_qr_scanner': 'QR Scanner',
  'cd_download': 'Download',
  'cd_upload': 'Upload',
  'cd_link': 'Link',
  'cd_copy': 'Copy',
  'cd_paste': 'Paste',
  'cd_phone': 'Phone',
  'cd_email': 'Email',
  'cd_lock': 'Lock',
  'cd_unlock': 'Unlock',
  'cd_key': 'Key',
  'cd_account': 'Account',
  'cd_logout': 'Log out',
  'cd_login': 'Log in',

  // Generic
  'cd_profile_photo': 'Profile photo',
  'cd_group_photo': 'Group photo',
  'cd_game_photo': 'Game photo',
  'cd_location_photo': 'Location photo',
  'cd_decorative': 'Decorative image',
  'cd_icon': 'Icon',
  'action_click': 'Click',
  'action_tap': 'Tap',
};

function getIconName(iconCode) {
  // Extrai o nome do ícone de padrões como:
  // Icons.Default.Add
  // Icons.Filled.Star
  // Icons.Outlined.Person
  const match = iconCode.match(/Icons\.(Default|Filled|Outlined|Rounded|Sharp|TwoTone|AutoMirrored\.Filled)\.(\w+)/);
  if (match) {
    return match[2];
  }
  return null;
}

function getContentDescriptionKey(iconCode) {
  const iconName = getIconName(iconCode);
  if (iconName && iconDescriptions[iconName]) {
    return iconDescriptions[iconName];
  }
  return 'cd_icon'; // fallback genérico
}

function fixIconsInFile(filePath, content) {
  let modified = content;
  let hasChanges = false;
  let needsStringResourceImport = false;

  // Pattern 1: Icon(Icons.xxx) sem contentDescription
  const iconPattern1 = /Icon\(\s*(Icons\.[^\)]+)\s*\)/g;
  modified = modified.replace(iconPattern1, (match, iconCode) => {
    if (!match.includes('contentDescription')) {
      hasChanges = true;
      needsStringResourceImport = true;
      const cdKey = getContentDescriptionKey(iconCode);
      return `Icon(${iconCode}, contentDescription = stringResource(R.string.${cdKey}))`;
    }
    return match;
  });

  // Pattern 2: Icon(\n    imageVector = Icons.xxx\n) sem contentDescription
  const iconPattern2 = /Icon\(\s*imageVector\s*=\s*(Icons\.[^,\)]+)([^)]*)\)/gs;
  modified = modified.replace(iconPattern2, (match, iconCode, rest) => {
    if (!match.includes('contentDescription')) {
      hasChanges = true;
      needsStringResourceImport = true;
      const cdKey = getContentDescriptionKey(iconCode);
      // Adiciona contentDescription após imageVector
      return `Icon(imageVector = ${iconCode}, contentDescription = stringResource(R.string.${cdKey})${rest})`;
    }
    return match;
  });

  // Pattern 3: Icon { Icon(xxx) } - IconButton com Icon dentro
  const iconButtonPattern = /IconButton[^{]*\{[^}]*Icon\(\s*(Icons\.[^\)]+)\s*\)[^}]*\}/gs;
  modified = modified.replace(iconButtonPattern, (match) => {
    if (!match.includes('contentDescription')) {
      const iconMatch = match.match(/Icon\(\s*(Icons\.[^\)]+)\s*\)/);
      if (iconMatch) {
        hasChanges = true;
        needsStringResourceImport = true;
        const cdKey = getContentDescriptionKey(iconMatch[1]);
        return match.replace(
          iconMatch[0],
          `Icon(${iconMatch[1]}, contentDescription = stringResource(R.string.${cdKey}))`
        );
      }
    }
    return match;
  });

  // Adiciona import se necessário
  if (needsStringResourceImport && !modified.includes('import androidx.compose.ui.res.stringResource')) {
    // Encontra a seção de imports e adiciona
    const importPattern = /(import androidx\.compose[^\n]*\n)/;
    if (importPattern.test(modified)) {
      modified = modified.replace(
        importPattern,
        '$1import androidx.compose.ui.res.stringResource\n'
      );
    } else {
      // Se não há imports do Compose, adiciona no início dos imports
      const packagePattern = /(package [^\n]+\n\n)/;
      if (packagePattern.test(modified)) {
        modified = modified.replace(
          packagePattern,
          '$1import androidx.compose.ui.res.stringResource\n\n'
        );
      }
    }
  }

  return { modified, hasChanges };
}

function fixImagesInFile(filePath, content) {
  let modified = content;
  let hasChanges = false;
  let needsStringResourceImport = false;

  // Pattern: AsyncImage sem contentDescription
  const asyncImagePattern = /AsyncImage\s*\(\s*model\s*=\s*([^,\)]+)([^)]*)\)/gs;
  modified = modified.replace(asyncImagePattern, (match, model, rest) => {
    if (!match.includes('contentDescription')) {
      hasChanges = true;
      needsStringResourceImport = true;
      // Determina o tipo de imagem baseado no contexto
      let cdKey = 'cd_image';
      if (filePath.includes('Profile') || model.includes('photoUrl') || model.includes('profilePicture')) {
        cdKey = 'cd_profile_photo';
      } else if (filePath.includes('Group') || model.includes('groupPhoto')) {
        cdKey = 'cd_group_photo';
      } else if (filePath.includes('Game') || model.includes('gamePhoto')) {
        cdKey = 'cd_game_photo';
      } else if (filePath.includes('Location') || model.includes('locationPhoto')) {
        cdKey = 'cd_location_photo';
      }

      return `AsyncImage(model = ${model}, contentDescription = stringResource(R.string.${cdKey})${rest})`;
    }
    return match;
  });

  // Pattern: Image(painter = ...) sem contentDescription
  const imagePattern = /Image\s*\(\s*painter\s*=\s*([^,\)]+)([^)]*)\)/gs;
  modified = modified.replace(imagePattern, (match, painter, rest) => {
    if (!match.includes('contentDescription')) {
      hasChanges = true;
      needsStringResourceImport = true;
      return `Image(painter = ${painter}, contentDescription = stringResource(R.string.cd_image)${rest})`;
    }
    return match;
  });

  // Adiciona import se necessário
  if (needsStringResourceImport && !modified.includes('import androidx.compose.ui.res.stringResource')) {
    const importPattern = /(import androidx\.compose[^\n]*\n)/;
    if (importPattern.test(modified)) {
      modified = modified.replace(
        importPattern,
        '$1import androidx.compose.ui.res.stringResource\n'
      );
    }
  }

  return { modified, hasChanges };
}

function fixClickablesInFile(filePath, content) {
  let modified = content;
  let hasChanges = false;
  let needsStringResourceImport = false;

  // Pattern: .clickable { ... } sem onClickLabel
  const clickablePattern = /\.clickable\s*\{/g;
  modified = modified.replace(clickablePattern, (match) => {
    // Verifica se já tem onClickLabel no contexto próximo
    const contextBefore = modified.substring(Math.max(0, modified.indexOf(match) - 200), modified.indexOf(match));
    if (!contextBefore.includes('onClickLabel')) {
      hasChanges = true;
      needsStringResourceImport = true;
      return '.clickable(onClickLabel = stringResource(R.string.action_click)) {';
    }
    return match;
  });

  // Pattern: .combinedClickable { ... } sem onClickLabel
  const combinedClickablePattern = /\.combinedClickable\s*\{/g;
  modified = modified.replace(combinedClickablePattern, (match) => {
    const contextBefore = modified.substring(Math.max(0, modified.indexOf(match) - 200), modified.indexOf(match));
    if (!contextBefore.includes('onClickLabel')) {
      hasChanges = true;
      needsStringResourceImport = true;
      return '.combinedClickable(onClickLabel = stringResource(R.string.action_click)) {';
    }
    return match;
  });

  // Adiciona import se necessário
  if (needsStringResourceImport && !modified.includes('import androidx.compose.ui.res.stringResource')) {
    const importPattern = /(import androidx\.compose[^\n]*\n)/;
    if (importPattern.test(modified)) {
      modified = modified.replace(
        importPattern,
        '$1import androidx.compose.ui.res.stringResource\n'
      );
    }
  }

  return { modified, hasChanges };
}

function processFile(filePath) {
  try {
    let content = fs.readFileSync(filePath, 'utf8');
    let totalChanges = false;

    // Fix Icons
    let result = fixIconsInFile(filePath, content);
    if (result.hasChanges) {
      content = result.modified;
      totalChanges = true;
    }

    // Fix Images
    result = fixImagesInFile(filePath, content);
    if (result.hasChanges) {
      content = result.modified;
      totalChanges = true;
    }

    // Fix Clickables
    result = fixClickablesInFile(filePath, content);
    if (result.hasChanges) {
      content = result.modified;
      totalChanges = true;
    }

    if (totalChanges) {
      fs.writeFileSync(filePath, content, 'utf8');
      return true;
    }

    return false;
  } catch (error) {
    console.error(`Error processing ${filePath}:`, error.message);
    return false;
  }
}

function getAllKotlinFiles(dir) {
  let files = [];
  const entries = fs.readdirSync(dir, { withFileTypes: true });

  for (const entry of entries) {
    const fullPath = path.join(dir, entry.name);
    if (entry.isDirectory()) {
      files = files.concat(getAllKotlinFiles(fullPath));
    } else if (entry.isFile() && entry.name.endsWith('.kt')) {
      files.push(fullPath);
    }
  }

  return files;
}

function generateStringsXml() {
  let xml = '\n    <!-- Content Descriptions for Accessibility -->\n';

  for (const [key, value] of Object.entries(contentDescriptionStrings)) {
    xml += `    <string name="${key}">${value}</string>\n`;
  }

  return xml;
}

function updateStringsXml(stringsPath) {
  try {
    let content = fs.readFileSync(stringsPath, 'utf8');

    // Verifica se já tem as strings
    if (content.includes('<!-- Content Descriptions for Accessibility -->')) {
      console.log('strings.xml já contém as content descriptions');
      return;
    }

    // Adiciona antes da tag </resources>
    const newStrings = generateStringsXml();
    content = content.replace('</resources>', `${newStrings}</resources>`);

    fs.writeFileSync(stringsPath, content, 'utf8');
    console.log('✅ strings.xml atualizado com content descriptions');
  } catch (error) {
    console.error('Erro ao atualizar strings.xml:', error.message);
  }
}

// Main execution
console.log('🚀 Iniciando correção automática de acessibilidade...\n');

const appDir = path.join(__dirname, '..', 'app', 'src', 'main', 'java');
const stringsPath = path.join(__dirname, '..', 'app', 'src', 'main', 'res', 'values', 'strings.xml');

// Step 1: Update strings.xml
console.log('Step 1: Atualizando strings.xml...');
updateStringsXml(stringsPath);

// Step 2: Process all Kotlin files
console.log('\nStep 2: Processando arquivos Kotlin...');
const kotlinFiles = getAllKotlinFiles(appDir);
let processedCount = 0;
let modifiedCount = 0;

for (const file of kotlinFiles) {
  processedCount++;
  if (processFile(file)) {
    modifiedCount++;
    console.log(`✓ ${path.relative(appDir, file)}`);
  }

  // Progress indicator
  if (processedCount % 50 === 0) {
    console.log(`Progresso: ${processedCount}/${kotlinFiles.length} arquivos processados...`);
  }
}

console.log(`\n✅ Concluído!`);
console.log(`📊 Estatísticas:`);
console.log(`   - Total de arquivos: ${kotlinFiles.length}`);
console.log(`   - Arquivos modificados: ${modifiedCount}`);
console.log(`   - Arquivos sem alterações: ${kotlinFiles.length - modifiedCount}`);
console.log(`\n🔍 Execute './gradlew compileDebugKotlin' para validar as mudanças`);
